### PR TITLE
TRD fixed charge trklt PID fix

### PR DESCRIPTION
**Pull request by martenole**
This works for the FST. I am also processing raw data at the moment, but it is very slow. Will report here when I get the results from the actual data.
Dynamic charge range still needs some work so that it can be handled properly.

### CODE REVIEW

**Comment by martenole** [comment]
I am going in a bit more detail through the TRD raw data handling. Starting here with the Sim->Raw conversion and will next more through the raw data reader. Comments are of course very welcome.

**Comment by bazinski** [comment]
@martenole 2 weeks and all that code is yours ;-)

**Comment by martenole** [comment]
Hi @bazinski and @tdietel 
this is ready for review now. Please let me know what you think. Apart from cleanup there are the following improvements:
- tracklet PID is consistent when going from sim to raw to extracted digits and tracklets
- the trigger offset is also stored in the half CRU header. Before it was not taken into account by the raw reader so that after the conversion sim to raw and reading the raw data the trigger times were shifted by 15 BCs. Now this is consistent
- trap2raw can also store the results per CRU and not only per half-CRU

File: Detectors/TRD/simulation/src/Trap2CRU.cxx [hunk 1]
```diff
@@ -682,149 +556,102 @@ void Trap2CRU::convertTrapData(o2::trd::TriggerRecord const& triggerrecord, cons
...
     mEndPointID = halfcru % 2; // just the upper or lower half of the cru, hence %2 of the the halfcru number.
-    std::string sside;
-    (side) ? sside = "C" : sside = "A";
-    //15 links per half cru or cru end point.
-    memset(&mRawData[0], 0, sizeof(mRawData[0]) * mRawData.size()); //   zero the rawdata storage
+    // 15 links per half cru or cru end point.
     HalfCRUHeader halfcruheader;
     //now write the cruheader at the head of all the data for this halfcru.
-    buildHalfCRUHeader(halfcruheader, triggerrecord.getBCData().bc, halfcru);
+    buildHalfCRUHeader(halfcruheader, ir.bc, halfcru);
```
Comment (martenole) [inline Detectors/TRD/simulation/src/Trap2CRU.cxx:563]: @bazinski here I am using the shifted trigger time

Comment (bazinski) [reply Detectors/TRD/simulation/src/Trap2CRU.cxx:563]: ah cool, I could not find it ;-)

**Comment by martenole** [comment]
Hi @TimoWilken 
seems that the o2-cs8 build failed on alibuild08.cern.ch due to no space left:
```
CMake Error at cmake/O2DataFile.cmake:57 (file):
  file COPY cannot copy file
  "/sw/SOURCES/O2/9373-slc8_x86-64/0/Common/maps/mfchebKGI_sym.root" to
  "/sw/BUILD/a34025f8969aef2ae9dde710a38dfc6031155eba/O2/stage/share/Common/maps/mfchebKGI_sym.root":
  No space left on device.
```
Could you please have a look?

**Comment by martenole** [comment]
similar issue on alinsure for fullCI
```
## sw/BUILD/O2-latest/log
--
[3005/3854] Building CXX object Detectors/GRP/workflows/CMakeFiles/O2lib-GRPCalibrationWorkflow.dir/src/GRPLHCIFfileSpec.cxx.o
FAILED: Detectors/GRP/workflows/CMakeFiles/O2lib-GRPCalibrationWorkflow.dir/src/GRPLHCIFfileSpec.cxx.o 
/sw/slc8_x86-64/GCC-Toolchain/v10.2.0-alice2-5/bin/c++ -DBOOST_ALL_NO_LIB -DBOOST_CONTAINER_DYN_LINK -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SERIALIZATION_DYN_LINK -DDPL_ENABLE_BACKTRACE -DENABLE_UPGRADES -DFAIRMQ_HAS_STD_FILESYSTEM=1 -DFMT_LOCALE -DFMT_SHARED -DO2lib_GRPCalibrationWorkflow_EXPORTS -DZDC_FASTSIM_ONNX -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Detectors/GRP/workflows/include -I/sw/BUILD/a34025f8969aef2ae9dde710a38dfc6031155eba/O2/Detectors/GRP/workflows -I/sw/BUILD/a34025f8969aef2ae9dde710a38dfc6031155eba/O2/Detectors/GRP/workflows/src -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Framework/Core/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Common/Utils/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/DataFormats/common/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Common/Constants/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/GPU/Common -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Framework/Logger/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Common/MathUtils/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/DataFormats/Headers/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/DataFormats/MemoryResources/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/CCDB/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Framework/Foundation/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Framework/Foundation/3rdparty/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Utilities/PCG/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Detectors/GRP/calibration/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Detectors/DCS/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Detectors/Calibration/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/DataFormats/Reconstruction/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/GPU/Utils -I/sw/SOURCES/O2/9373-slc8_x86-64/0/DataFormats/Detectors/Common/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Utilities/rANS/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Detectors/Base/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Common/Field/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/DataFormats/Parameters/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Common/Types/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/Common/SimConfig/include -I/sw/SOURCES/O2/9373-slc8_x86-64/0/DataFormats/Calibration/include -isystem /sw/slc8_x86-64/Common-O2/v1.6.0-47/include -isystem /sw/slc8_x86-64/boost/v1.75.0-44/include -isystem /sw/slc8_x86-64/Configuration/v2.6.3-35/include -isystem /sw/slc8_x86-64/libInfoLogger/v2.5.0-1/include -isystem /sw/slc8_x86-64/Monitoring/v3.12.0-17/include -isystem /sw/slc8_x86-64/curl/7.70.0-1/include -isystem /sw/slc8_x86-64/FairMQ/v1.4.51-5/include/fairmq -isystem /sw/slc8_x86-64/FairMQ/v1.4.51-5/include -isystem /sw/slc8_x86-64/FairLogger/v1.11.1-1/include -isystem /sw/slc8_x86-64/fmt/8.1.1-6/include -isystem /sw/slc8_x86-64/ROOT/v6-26-04-patches-alice1-1/include -isystem /sw/slc8_x86-64/FairRoot/v18.4.8-6/include -isystem /sw/slc8_x86-64/VMC/v1-1-p1-31/include/vmc -isystem /sw/slc8_x86-64/arrow/v5.0.0-alice3-34/include -isystem /sw/slc8_x86-64/Vc/1.4.1-25/include -isystem /sw/slc8_x86-64/ms_gsl/4.0.0-8/include -isystem /sw/slc8_x86-64/libjalienO2/0.1.3-22/include -isystem /sw/slc8_x86-64/RapidJSON/v1.1.0-alice2-28/include -isystem /sw/slc8_x86-64/libuv/v1.40.0-25/include -isystem /sw/slc8_x86-64/TBB/v2021.5.0-17/include -fPIC -O2 -std=c++17 -O2 -g -DNDEBUG -Wno-unknown-warning-option -fPIC -std=gnu++17 -MD -MT Detectors/GRP/workflows/CMakeFiles/O2lib-GRPCalibrationWorkflow.dir/src/GRPLHCIFfileSpec.cxx.o -MF Detectors/GRP/workflows/CMakeFiles/O2lib-GRPCalibrationWorkflow.dir/src/GRPLHCIFfileSpec.cxx.o.d -o Detectors/GRP/workflows/CMakeFiles/O2lib-GRPCalibrationWorkflow.dir/src/GRPLHCIFfileSpec.cxx.o -c /sw/SOURCES/O2/9373-slc8_x86-64/0/Detectors/GRP/workflows/src/GRPLHCIFfileSpec.cxx
/sw/SOURCES/O2/9373-slc8_x86-64/0/Detectors/GRP/workflows/src/GRPLHCIFfileSpec.cxx:256:1: fatal error: error writing to /tmp/ccsd6EBb.s: No space left on device
  256 | } // namespace o2
      | ^
compilation terminated.
```

**Comment by TimoWilken** [comment]
@martenole sorry about that, working on it!

**Comment by martenole** [comment]
No problem, thanks a lot Timo

**Review by bazinski — APPROVED** [review]
Wow a lot of old junk in there :-)

**Comment by davidrohr** [comment]
@martenole : just checing this locally in FST. Should we merge it for tomorrow's software, or wait?

**Comment by martenole** [comment]
Hi @davidrohr , thanks a lot. Actually, yes, if your tests turn out OK it would be nice to have this merged before tomorrow. The only change for the raw data decoding is the different and hopefully now correct interpretation of PID values for the tracklets.
Cheers,
Ole

**Pull request merged**